Automattic/jetpack
# AI Assistant: Extend list block and enhance content parsing

Fixes #31875
Fixes #31125
Part of #31878

## Proposed changes:

* Adds `core/list` to the extended blocks
* Breaks down the `useTextContentFromSelectedBlocks` hook in two for better separation of concerns
* Fixes the broken lists by processing the generated markdown
* Removes the dependency on a blocks `content` attribute, reading its HTML content directly instead

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Add a list and some paragraphs
* Check that you can use the `AI Assistant` toolbar menu on lists
* Check that, when transforming multiple paragraphs, their content is not concatenated into one paragraph, remaining separate

https://github.com/Automattic/jetpack/assets/8486249/e24b6618-5faa-4d72-8be1-32e231056a4c
